R/html_vignette.R

```diff
@@ -19,25 +19,42 @@
 #'
 #' @inheritParams html_document
 #' @param ... Additional arguments passed to \code{\link{html_document}}
+#' @param readme Use this vignette as the package README.md file (i.e. render
+#'   it as README.md to the package root)
 #' @return R Markdown output format to pass to \code{\link{render}}
 #' @export
 html_vignette <- function(fig_width = 3,
                           fig_height = 3,
                           dev = 'png',
                           css = NULL,
+                          readme = FALSE,
                           ...) {
 
   if (is.null(css)) {
     css <- system.file("rmarkdown", "templates", "html_vignette" ,"resources",
       "vignette.css", package = "rmarkdown")
   }
 
-  html_document(fig_width = fig_width,
-                fig_height = fig_height,
-                dev = dev,
-                fig_retina = NULL,
-                css = css,
-                theme = NULL,
-                highlight = "pygments",
-                ...)
+  pre_knit <- function(input, ...) {
+    if (readme) {
+      rmarkdown::render(input,
+                        output_format = "github_document",
+                        output_file = "README.md",
+                        output_dir = dirname(dirname(input)))
```

Comment:
Is the intention here that this would only get called on vignettes in e.g. the `inst/doc` folder? (Ie, this would get called after e.g. `R CMD build` copies from `vignettes/` to `inst/doc/`?)

Just want to make sure that this wouldn't inadvertently leak out of the project folder for certain cases.